### DISCUSSION TITLE
Add IP with python custom script

### DISCUSSION THREAD

**Discussion by SRF-rueeggdo** [netbox-community/netbox#15482] [Q&A] [answered]
Hello everyone

I am trying to create a custom script to automatically detect specific devices and assign them to an IP address but I'm running in some errors.
This is my sample code:

```
from ipam.models import IPAddress
import ipaddress

def checkAndSaveLoop(self, loopIps):
    ip = "10.120.23.193/26"
    ip = ipaddress.IPv4Network(ip) #option 3 and 4
    newLoopIp = IPAddress(
    address=ip,
    status=IPAddressStatusChoices.STATUS_ACTIVE
    )
    newLoopIp.clean()
    newLoopIp.save()
```

I have tried several options:

1. pass IP as string (without mask)
Exception: AttributeError: 'str' object has no attribute 'prefixlen'
2. pass IP as string (with mask)
Exception: AttributeError: 'str' object has no attribute 'prefixlen'
3. pass IP as ipaddress.IPv4Network object (without mask)
Exception: AttributeError: 'IPv4Network' object has no attribute '_value'
4. pass IP as ipaddress.IPv4Network object (with mask)
Exception: ValueError: ip has host bits set

Does anyone have experience of how to create an IP for a custom script?

Netbox v3.6.4

**Comment by candlerb** [comment, marked as answer]
Netbox uses the netaddr library, not ipaddress. You can find examples in the test suite, e.g. `netbox/ipam/tests/test_ordering.py`

```
import netaddr
...
IPAddress(status=IPAddressStatusChoices.STATUS_ACTIVE, vrf=vrf1, address=netaddr.IPNetwork('10.0.0.1/24')),
```
